packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx

```diff
@@ -68,62 +68,69 @@ export function usePreviewSrc(
   return [src, loading]
 }
 
-export const ImageField = wrapFieldsWithMeta<InputProps, ImageProps>(
-  (props) => {
-    const cms = useCMS()
-    const { form, field } = props
-    const { name, value } = props.input
-    const [src, srcIsLoading] = usePreviewSrc(
-      value,
-      name,
-      form.getState().values,
-      field.previewSrc
-    )
-
-    let onClear: any
-    if (props.field.clearable) {
-      onClear = () => props.input.onChange('')
-    }
+export const ImageField = wrapFieldsWithMeta<InputProps, ImageProps>(props => {
+  const cms = useCMS()
+  const { form, field } = props
+  const { name, value } = props.input
+  const [src, srcIsLoading] = usePreviewSrc(
+    value,
+    name,
+    form.getState().values,
+    field.previewSrc
+  )
+
+  let onClear: any
+  if (props.field.clearable) {
+    onClear = () => props.input.onChange('')
+  }
 
-    function onChange(media?: Media) {
-      if (media) {
-        props.input.onChange('')
-        props.input.onChange(media)
-      }
+  function onChange(media?: Media) {
+    if (media) {
+      const mediaParse = cms.media.store
+
+      const parsedValue =
+        // @ts-ignore
+        typeof cms?.media?.store?.parse === 'function'
+          ? // @ts-ignore
+            cms.media.store.parse(media)
+          : // @ts-ignore
+            mediaParse.parse(media)
```

Comment:
Trying to make sense of this... if `cms.media.store.parse` is a `function`, then use it.  If not... use `cms.media.store.parse`?  Might be missing the point here, but this almost reads like we are sourcing the same `parse()` method.

Reply:
That's a mistake, I meant to just pass `media` for the fallback condition. Will fix